app.js

```diff
@@ -1,97 +1,140 @@
-    document.getElementById('hide-button').style.display = 'block';
+  const paragraphs = document.querySelectorAll('#projects_items .load_more');
+  for (let i = 0; i < paragraphs.length; i++) {
+    paragraphs[i].classList.add('show');
+  }
+  const hr = document.querySelectorAll('.line_hr');
+  for (let i = 0; i < hr.length; i++) {
+    hr[i].classList.add('sh');
+  }
+  document.getElementById('load-more-button').style.display = 'none';
+  document.getElementById('hide-button').style.display = 'block';
 }
 
 function hide() {
-    const paragraphs = document.querySelectorAll('#projects_items .load_more');
-    for (let i = 0; i < paragraphs.length; i++) {
-      paragraphs[i].classList.remove('show');
-    }
-    const hr = document.querySelectorAll('.line_hr');
-    for (let i = 0; i < hr.length; i++) {
-        hr[i].classList.remove('sh');
-      }
-    document.getElementById('load-more-button').style.display = 'block';
-    document.getElementById('hide-button').style.display = 'none';
+  const paragraphs = document.querySelectorAll('#projects_items .load_more');
+  for (let i = 0; i < paragraphs.length; i++) {
+    paragraphs[i].classList.remove('show');
+  }
+  const hr = document.querySelectorAll('.line_hr');
+  for (let i = 0; i < hr.length; i++) {
+    hr[i].classList.remove('sh');
+  }
+  document.getElementById('load-more-button').style.display = 'block';
+  document.getElementById('hide-button').style.display = 'none';
 }
 
-const members_url =
-  "https://api.github.com/orgs/move-fast-and-break-things/members";
+const members_url = "https://api.github.com/orgs/move-fast-and-break-things/members";
 
+// fetch authors data
 async function getAuthors(apiURL) {
-  let authors = [];
-  const result = await fetch(apiURL); // Using github api to get the list of members
-  const data = await result.json();
-  for (const element of data) {
-    // Getting avatar name, biography and url from each user, then putting them into an array
-    const result = await fetch(element.url);
-    const data = await result.json();
-    authors.push({
-      name: data.name ? data.name : data.login,
-      bio: data.bio,
-      avatar: data.avatar_url,
-      html_url: data.html_url,
-    });
+  try {
+    const response = await fetch(apiURL);
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+    const data = await response.json();
+    return Promise.all(data.map(async (element) => {
+      const userResponse = await fetch(element.url);
+      if (!userResponse.ok) {
+        throw new Error(`HTTP error! status: ${userResponse.status}`);
+      }
+      const userData = await userResponse.json();
+      return {
+        name: userData.name || userData.login,
+        bio: userData.bio,
+        avatar: userData.avatar_url,
+        html_url: userData.html_url,
+      };
+    }));
+  } catch (error) {
+    console.error("Error fetching authors:", error);
+    return [];
```

Comment:
let's let this one crash. if fetching fails, it fails. let the outer code handle it
```suggestion
  const response = await fetch(apiURL);
  if (!response.ok) {
    throw new Error(`HTTP error! status: ${response.status}`);
  }
  const data = await response.json();
  return Promise.all(data.map(async (element) => {
    const userResponse = await fetch(element.url);
    if (!userResponse.ok) {
      throw new Error(`HTTP error! status: ${userResponse.status}`);
    }
    const userData = await userResponse.json();
    return {
      name: userData.name || userData.login,
      bio: userData.bio,
      avatar: userData.avatar_url,
      html_url: userData.html_url,
    };
  }));
```